src/main/java/org/zalando/nakadi/domain/BatchFactory.java

```diff
@@ -13,10 +13,35 @@
         int brackets = 0;
         boolean insideQuote = false;
         boolean escaped = false;
-        if ((!events.startsWith("[")) || (!events.endsWith("]"))) {
-            throw new JSONException("Array must be surrounded with square brackets");
+        int start = 0;
+        final int length = events.length();
+        int end = length - 1;
+
+        while ((events.charAt(start) == ' '
+                || events.charAt(start) == '\t'
+                || events.charAt(start) == '\n'
+                || events.charAt(start) == '\r')
+                && start < end) {
+            start++;
+        }
+        while ((events.charAt(end) == ' '
```

Comment:
If this condition is extracted to its own method, then this should be refactored to reuse it.